src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -114,33 +113,19 @@ public void clusterChanged(ClusterChangedEvent event) {
                 }
             }
 
-            // check and write changes in indices
-            for (IndexMetaData indexMetaData : newMetaData) {
-                String writeReason = null;
-                IndexMetaData currentIndexMetaData;
-                if (currentMetaData == null) {
-                    // a new event..., check from the state stored
-                    try {
-                        currentIndexMetaData = metaStateService.loadIndexState(indexMetaData.index());
```

Comment:
as far as I am concerned we could just skip it since we write the new one anyway. Yet, we might have a new meta on disk but that would be a bug too no? so I think we can drop it?